### ISSUE TITLE
Update integration-cli tests

### ISSUE THREAD

**Issue by vieux** [moby/moby#32866]
Many tests rely today on the Docker CLI.
Since the CLI is moving out of this repo (see https://github.com/moby/moby/pull/32694), we need to
- convert tests to use the API

or

- move tests somewhere else

**Comment by cpuguy83** [comment]
We should probably update the contributing guidelines to indicate that users should not be using integration-cli. Or at the least only make API tests.

To go along with this either docker/cli needs to setup integration testing or some separate integration repo for "docker-ce".

**Comment by vdemeester** [comment]
@cpuguy83 yes and we need to setup a ci check with those for the time being. I'll start working on that next week 😇

**Comment by vdemeester** [comment]
Note that there is discussion happening in https://forums.mobyproject.org/t/evolution-of-testing-moby/38

**Comment by dnephin** [comment]
Opened #34407 to document the proposal from the forum thread.

**Comment by dnephin** [comment]
Current status: #34000 and #34407 have merged so we have a new suite and a document for testing guidelines. A validation check has been added to prevent any new cli tests from being added (#34552).

Ongoing: removing cli-only tests: #34560 and #34548

Next steps:
* continue removing any cli-only tests
* move more tests to the new suite to provide more examples


I'll update the description of this issue as well.

**Comment by michael2012z** [comment]
Hello, everyone,

What is the latest status of this task? I saw there had been quite a few PR's working on migrating legacy CLI test cases into API cases. But the CLI cases are still the majority of all integration test. Is this job still calling for contribution? Just want to know if I can help in this issue.
